rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala

```diff
@@ -100,18 +100,21 @@ abstract class RSpaceOps[C, P, A, R, K](val store: IStore[C, P, A, K], val branc
       store.withTrieTxn(txn) { trieTxn =>
         store.trieStore.validateAndPutRoot(trieTxn, store.trieBranch, root)
         val leaves = store.trieStore.getLeaves(trieTxn, root)
+        eventLog.update(const(Seq.empty))
+        store.clearTrieUpdates()
         store.clear(txn)
         restoreInstalls(txn)
         store.bulkInsert(txn, leaves.map { case Leaf(k, v) => (k, v) })
       }
     }
 
-  override def clear(): Unit =
-    store.withTxn(store.createTxnWrite()) { txn =>
-      eventLog.update(const(Seq.empty))
-      store.clearTrieUpdates()
-      store.eventsCounter.reset()
-      store.clear(txn)
-      restoreInstalls(txn)
-    }
+  override def clear(): Unit = {
+    val emptyRootHash: Blake2b256Hash =
+      store.withTxn(store.createTxnRead()) { txn =>
+        store.withTrieTxn(txn) { trieTxn =>
+          store.trieStore.getAllPastRoots(trieTxn).last
```

Comment:
just a thought: since trieStore owns the PastRoots abstraction and the emptyRoot is a special boi maybe it should be treated in a special way? perhaps instead of: `getAllPastRoots(trieTxn).last` a `getEmptyRoot(trieTxn)`?

Reply:
I had a similar thought.  It would demand adding another sub-database and method(s), so I thought it best to stick with a simple approach for this fix, but I think we should consider ticketing a more robust solution like this.